classes/class-wc-connect-taxjar-integration.php

```diff
@@ -286,8 +286,22 @@ public function _error( $message ) {
 		$formatted_message = is_scalar( $message ) ? $message : json_encode( $message );
 
 		//ignore error messages caused by customer input
-		if ( 1 === preg_match( '/to_zip .* is not used within to_state/', $formatted_message ) ) {
-			wc_add_notice( __( 'ZIP does not match the selected state.', 'woocommerce-services' ), 'error' );
+		$state_zip_mismatch = 1 === preg_match( '/to_zip [^"}]* is not used within to_state/', $formatted_message );
```

Comment:
Since the error message is something like "to_zip 20112 is not used within to_state AL" I don't see a reason to try and detect the end of the JSON encoded string. Can we just look for alphanumeric characters and whatever other special characters are allowed for postcodes? Something like `[\w\-]+`?

Reply:
Perhaps we're overthinking it. I'm really just trying to keep an eye on performance. Since this isn't critical functionality, what do you think about using `strpos()` instead of `preg_match()` for both checks since it's faster?